lib/charms/traefik_route_k8s/v0/traefik_route.py

```diff
@@ -0,0 +1,174 @@
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+r"""# Interface Library for traefik_route.
+
+This library wraps relation endpoints for traefik_route. The requirer of this
+relation is a charm in need of ingress (or a proxy thereof), the
+provider is the traefik-k8s charm.
+
+## Getting Started
+
+To get started using the library, you just need to fetch the library using `charmcraft`.
+
+```shell
+cd some-charm
+charmcraft fetch-lib charms.traefik_route_k8s.v0.traefik_route
+```
+
+```yaml
+requires:
+    traefik_route:
+        interface: traefik_route
+        limit: 1
+```
```

Comment:
As I understand it - the design was such that the workload needing ingress would only ever need to communicate on the `ingress` or `ingress_per_unit` relation. 

This interface should only be present between traefik-k8s and traefik-route-k8s, right?